elastic/kibana
# [Security Solution][Endpoint] Add agent status service client

## Summary

Agent status service and client for fetching endpoint and third party agent statuses.

**To test**
1. setup ES/Kibana
2. Set up a sentinel one connector using `x-pack/plugins/security_solution/scripts/endpoint/run_sentinelone_host.js` script.
3. Verify that the detection rule is created. It should be prefixed with `Test Rule -`.
4. Shell into the newly created sentinel one agent and do a `nslookup`.
5. Go to alerts page and wait until a malware alert shows up from the sentinel one agent.
6. Click and open the alert to see Alert Details flyout.
7. You should see the agent's status as `Healthy` under the `Highlighted fields` section. 
8. Click on `Take action' and `isolate` the host.
9. Verify that the agent shows an `Isolated` badge in addition to `Healthy` after some time.

### Checklist


### For maintainers
